### PR TITLE
[INLONG-3158][Audit] Proxy support TubeMQ

### LINKED ISSUE #3158: [Feature][Audit] Proxy support TubeMQ
### Description

_No response_

### Use case

_No response_

### Are you willing to submit PR?

- [ ] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by pocozh**
### Title Name: [INLONG-3158][Audit] Proxy support TubeMQ

where *XYZ* should be replaced by the actual issue number.

Fixes #3158 

### Modifications

- add TubeMQ sink
- modify proxy-start.sh, to support both pulsar and tubeMQ

### CODE REVIEW

File: inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/TubeSink.java [hunk 1]
```diff
@@ -0,0 +1,660 @@
...
+            } catch (Throwable e) {
+                logger.error("destroy producer error in tubesink, ex {}", e.getMessage());
+            }
+        }
+        producerMap.clear();
+
+        if (sessionFactory != null) {
+            try {
+                sessionFactory.shutdown();
+            } catch (TubeClientException e) {
```
Comment (healchow) [inline inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/TubeSink.java:416]: If different exceptions are handled the same way, using one `catch (Exception e)` could be better.

File: inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/TubeSink.java [hunk 2]
```diff
@@ -0,0 +1,660 @@
...
+     * @throws TubeClientException
+     */
+    private void initTopicProducer(String topic) throws TubeClientException {
+        if (StringUtils.isEmpty(topic)) {
+            logger.error("topic is empty");
+            return;
+        }
+        if (sessionFactory == null) {
+            logger.error("sessionFactory is null, can't create producer");
+            return;
```
Comment (healchow) [inline inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/TubeSink.java:440]: Maybe need to throw an exception when the sessionFactory is null?

File: inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/TubeSink.java [hunk 3]
```diff
@@ -0,0 +1,660 @@
...
+                        if (es != null) {
+                            event = es.getEvent();
+                            if (event.getHeaders().containsKey(TOPIC)) {
+                                topic = event.getHeaders().get(TOPIC);
+                            }
+                        }
+                    } else {
+                        event = eventQueue.take();
+                        es = new EventStat(event);
+//                            sendCnt.incrementAndGet();
```
Comment (healchow) [inline inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/TubeSink.java:521]: If not needed, remove it.

File: inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/TubeSink.java [hunk 4]
```diff
@@ -0,0 +1,660 @@
...
+
+                    decrementFlag = true;
+
+                } catch (InterruptedException e) {
+                    logger.info("Thread {} has been interrupted!", Thread.currentThread().getName());
+                    return;
+                } catch (Throwable t) {
+                    if (t instanceof TubeClientException) {
+                        String message = t.getMessage();
+                        if (message != null && (message.contains("No available queue for topic")
```
Comment (healchow) [inline inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/TubeSink.java:566]: Magic strings are not recommended.

File: inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/TubeSink.java [hunk 5]
```diff
@@ -0,0 +1,660 @@
...
+        private long sendTime;
+
+        public MyCallback(org.apache.inlong.audit.sink.EventStat eventStat) {
+            this.myEventStat = eventStat;
+            this.sendTime = System.currentTimeMillis();
+        }
+
+        @Override
+        public void onMessageSent(final MessageSentResult result) {
+            if (result.isSuccess()) {
```
Comment (healchow) [inline inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/TubeSink.java:626]: It is recommended to use guard statements to reduce the cyclomatic complexity of the method.

File: inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/TubeSink.java [hunk 6]
```diff
@@ -0,0 +1,660 @@
...
+        sinkThreadPool = new Thread[threadNum];
+        eventQueue = new LinkedBlockingQueue<Event>(EVENT_QUEUE_SIZE);
+
+        diskIORatePerSec = context.getLong("disk-io-rate-per-sec", 0L);
+        if (diskIORatePerSec != 0) {
+            diskRateLimiter = RateLimiter.create(diskIORatePerSec);
+        }
+
+        linkMaxAllowedDelayedMsgCount = context.getLong(ConfigConstants.LINK_MAX_ALLOWED_DELAYED_MSG_COUNT,
+                80000L);
```
Comment (baomingyu) [inline inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/TubeSink.java:298]: make such default value as default static class field.

**Review by baomingyu — CHANGES_REQUESTED** [review]

File: inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/TubeSink.java (hunk 6 above)
Comment (pocozh) [reply inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/TubeSink.java:298]: done

File: inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/TubeSink.java (hunk 5 above)
Comment (pocozh) [reply inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/TubeSink.java:626]: done

**Review by baomingyu — APPROVED** [review]
+1

**Review by healchow — APPROVED** [review]

**Pull request merged**